akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala

```diff
@@ -148,14 +138,14 @@ abstract class EventSourcedBehavior[Command, Event, State >: Null] private[akka]
 
   def eventAdapter(): EventAdapter[Event, _] = NoOpEventAdapter.instance[Event]
 
-  def retentionCriteria: RetentionCriteria = RetentionCriteria()
+  def retentionCriteria: RetentionCriteria = RetentionCriteria.default
 
   /**
    * INTERNAL API: DeferredBehavior init
    */
   @InternalApi override def apply(context: typed.TypedActorContext[Command]): Behavior[Command] = {
     val snapshotWhen: (State, Event, Long) => Boolean = { (state, event, seqNr) =>
-      val n = snapshotEvery()
+      val n = retentionCriteria.snapshotEveryNEvents
```

Comment:
The default should still be to not automatically save snapshots. Now I think the default is `snapshotEveryNEvents=1000`. Perhaps rename `RetentionCriteria.default` to `RetentionCriteria.disabled` and use `snapshotEveryNEvents=0` there.

Reply:
Good. I did indeed think snapshotting was by default enabled for most use cases. Updating.